### ISSUE TITLE
Possible Jacobian bug 

### ISSUE THREAD

**Issue by mjmcnelis** [mjmcnelis/RKM.jl#43]
Should I not reuse the `i` index for adding 1 to the diagonal elements?
```
if root_method isa FixedPoint
                    @.. dy[:,i] -= error
                elseif root_method isa Newton
                    # evaluate current Jacobian
                    evaluate_system_jacobian!(jacobian_method, FE, J,
                                              ode_wrap!, y_tmp, f_tmp)

                    @.. J = J * (-A_T[i,i]*dt)          # J <- I - A.dt.J
                    for i in diagind(J)
                        J[i] += 1.0
                    end

                    # pass Jacobian and residual error to linear cache
                    linear_cache.A = J
                    linear_cache.b = error
                    solve!(linear_cache)
                    @.. dy[:,i] -= linear_cache.u
                end
```

**Comment by mjmcnelis** [comment]
Doesn't seem like it causes any issues but probably best to change it to avoid confusion

**Comment by mjmcnelis** [comment]
Closed w/ commit 77575e34208c3fa65a2c1bd0a753ecf9dc7e3778
